sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -315,16 +325,18 @@ def __init__(self, key, secret_uri, label=None, **kwargs):
         self.last_modified = kwargs.get("last_modified", None)
         self.read_only = kwargs.get("read_only", None)
         self.tags = kwargs.get("tags", {})
-        self.value = {"secret_uri": self._secret_uri}
+        self.value = secret_id
+        if not isinstance(secret_id, dict):
+            self.value = {"secret_uri": secret_id}
 
     @property
-    def secret_uri(self):
+    def secret_id(self):
         # type: () -> str
         self._validate()
         return self.value['secret_uri']
 
-    @secret_uri.setter
-    def secret_uri(self, value):
+    @secret_id.setter
+    def secret_id(self, value):
```

Comment:
why value needs to be a dict?

Reply:
This is how it is represented in the portal